samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt

```diff
@@ -0,0 +1,187 @@
+set(NANOPB_REPO_TAG "nanopb-0.4.1")
+
+# Check local folder for protoc and nanopb binaries presence
+file(GLOB_RECURSE LOCAL_PROTOC_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc")
+if (NOT LOCAL_PROTOC_BIN)
+    file(GLOB_RECURSE LOCAL_PROTOC_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc.exe")
+endif()
+file(GLOB_RECURSE LOCAL_NANOPB_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc-gen-nanopb*")
+
+# Check system for protoc and nanopb binaries presence
+find_program(PROTOC_BIN protoc)
+find_program(NANOPB_BIN protoc-gen-nanopb)
+
+# Prefer local binaries to make version matching with nanopb source repo easier
+if (LOCAL_PROTOC_BIN)
+    set(PROTOC_BIN ${LOCAL_PROTOC_BIN})
+endif()
+
+if (LOCAL_NANOPB_BIN)
+    set(NANOPB_BIN ${LOCAL_NANOPB_BIN})
+endif()
+
+if ((NOT NANOPB_BIN) OR (NOT PROTOC_BIN))
+    message(STATUS "nanopb not found.")
+    message(STATUS "Please download ${NANOPB_REPO_TAG} from here https://jpa.kapsi.fi/nanopb/download/")
+    message(STATUS "and extract to either this sample directory or somewhere cmake can find it via system PATH.")
+    message(FATAL_ERROR "${NANOPB_REPO_TAG} required by this project.")
+endif()
+
+message(STATUS "Using protoc: ${PROTOC_BIN}")
+message(STATUS "Using protoc-gen-nanopb: ${NANOPB_BIN}")
+message(STATUS "Using nanopb source tag: ${NANOPB_REPO_TAG}")
+
+function(compile_protobufs_in_dir dir recursive)
+    # Find all .proto files and include directories
+    if (recursive)
+        file(GLOB_RECURSE PROTO_FILES "${dir}/*.proto")
+    else()
+        file(GLOB PROTO_FILES "${dir}/*.proto")
+    endif()
+
+    foreach(file ${PROTO_FILES})
+        get_filename_component(FILE_PATH ${file} DIRECTORY)
+        list(APPEND PROTOC_INC_DIRS "-I${FILE_PATH}")
+    endforeach()
+
+    # Compile .proto files in their respective directories
+    foreach(file ${PROTO_FILES})
+        # Add all directories containing .proto files.
+        # Duplicates and current directory is then removed from "-I" arguments
+        set(PROTOC_ARGS "${PROTOC_INC_DIRS}")
+        get_filename_component(FILE_PATH ${file} DIRECTORY)
+        get_filename_component(FILE_NAME ${file} NAME)
+        list(REMOVE_ITEM PROTOC_ARGS "-I${FILE_PATH}")
+        list(REMOVE_DUPLICATES PROTOC_ARGS)
+        list(APPEND PROTOC_ARGS "-I.")
+        list(APPEND PROTOC_ARGS "--nanopb_out=${FILE_PATH}")
+        list(APPEND PROTOC_ARGS "${FILE_NAME}")
+        message(STATUS "Compiling proto definition: ${FILE_NAME}")
+        execute_process(
+            COMMAND ${PROTOC_BIN} ${PROTOC_ARGS}
+            WORKING_DIRECTORY ${FILE_PATH}
+        )
+    endforeach()
+endfunction()
+
+include(FetchContent)
+
+# Get repo containing protobuf definitions
+FetchContent_Declare(
+    gadgets_repo
+    GIT_REPOSITORY ${GADGETS_PROTO_REPO}
+    GIT_TAG        ${GADGETS_PROTO_TAG}
+)
+
+FetchContent_GetProperties(gadgets_repo)
+if(NOT gadgets_repo_POPULATED)
+    FetchContent_Populate(gadgets_repo)
+endif()
```

Comment:
we should not dowload code as part of CMake (or build).

This should go into the manifest file instead.